### DISCUSSION TITLE
Do not allow Chinese characters when packaging Android versions?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2906] [Q&A] [answered]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/91530077/5a896892-be29-43f7-95fa-14fc515649e8)
When I try to package Android, if there are Chinese characters in the added resources, I will report an error.
Is there any way to allow packaging of files with Chinese names?
If I try to download from the server to my phone's SD card using the server method, I am allowed to use Chinese, but when I package the original APK with a Chinese name, will an error be reported?

**Comment by D5CN** [comment]
yes.and this is just happen in PC.
in MAC you can use chinese in app-xml.
But don't worry,you can use UNICODE to instead.like this.
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/16438099/e672775a-36a8-4473-98c5-eea98d41830f)

是的，在PC上的确这样，但是使用MAC编译没有这个问题
解决它也比较简单，只要把中文转成UNICODE就可以了
我也是突发奇想，自己试过以后才发现的。。。

**Reply by Jason0247** [reply]
哈哈，看到了一个用中文的，那我就不费力了。
我在PC上打包时会出现这个问题，是文件夹不允许中文命名和文件也不允许，不是描述文件。

**Reply by D5CN** [reply]
嗯，这个不止在PC上，在MAC上同样不支持。
但是这种问题应该问题不大，因为不暴露给用户。只是自己改名字有点麻烦。

yes,not only in PC,it also not support chinese file name in mac.
but this filename will not show to enduser,so you just need rename the file.a lot of work^^

**Comment by ajwfrost** [comment]
@Jason0247 can I check, are you setting your application name/filename to have Chinese characters, or are you trying to include additional subfolders that have Chinese characters in the name? (or both?)

From what I recall, the app descriptor `<filename>` element needs to use ascii characters, but `<name>` can be utf8 and contain non-latin characters..

If you can zip up a quick test project and upload it here, we can take a look and see at what point the tools are complaining about it, and whether there are workarounds..

thanks

**Reply by Jason0247** [reply]
It is not a problem to use Chinese for the application name, and it is not a problem to use Chinese for the packaged APK name. However, if the packaging material includes Chinese naming, an error will be reported.
This is the engineering file I tested. I packaged it into a text file and named it in Chinese, so it is not allowed to be packaged. Regarding this issue, I have to use AIP to compress the file and then extract it into the applicationStorageDirectory directory of the application during runtime, which is very troublesome.
thanks
[test.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/13450671/test.zip)

**Reply by ajwfrost** [reply, marked as answer]
Okay thanks. So the initial problem is a restriction that was added because of Android's restrictions when using AAPT, but we no longer use that tool when packaging via Gradle.

So if we update that restriction so it allows through the Chinese filename, the initial APK is created properly .. it looks like the signing process subsequently screws this up.

In fact, yes if you take an APK that has the Chinese filename within the assets, and call "apksigner" on it from the Android SDK, the filename gets changed into a malformed format.... Although, it works when you use Android Studio to do this!

We'll have to look into this... the reason we switched to calling "apksigner" was to work around the issues with the internal signing code in ADT that doesn't work in JRE17+ .. we're likely to have to rework ADT anyway because of the issues with code-signing for APKs now that Animate have adopted JRE17, so we'll try to address this at the same time.

thanks
